support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButton.tsx

```diff
@@ -193,8 +190,11 @@ const onComplete = (res: PaymentResult) => {
 
 function updateTotal(props: PropTypes) {
 	// When the other tab is clicked, the value of amount is NaN
-	if (!Number.isNaN(props.amount) && props.paymentRequestObject) {
-		props.paymentRequestObject.update({
+	if (
+		!Number.isNaN(props.amount) &&
+		props.paymentRequestObject.status === 'AVAILABLE'
```

Comment:
Adding the tagged union hasn't added to much extra code. The main difference is that where we were checking that `props.paymentRequestObject` wasn't `null`, we now have to check `props.paymentRequestObject.status === 'AVAILABLE'`. It's more verbose, but I'd say the trade off is worth it

Reply:
👍